src/deserialization.hpp

```diff
@@ -93,123 +247,28 @@ class ConcreteTensorProtoDeserializator {
-                size_t i = 0;
-                for (auto& number : requestInput.contents().uint_contents()) {
-                    ptr[i++] = *(const_cast<uint8_t*>(reinterpret_cast<const uint8_t*>(&number)));
-                }
-                return tensor;
-                break;
-            }
-            case ovms::Precision::U16: {
-                ov::Tensor tensor = makeTensor(requestInput, tensorInfo);
-                uint16_t* ptr = reinterpret_cast<uint16_t*>(tensor.data());
-                size_t i = 0;
-                for (auto& number : requestInput.contents().uint_contents()) {
-                    ptr[i++] = *(const_cast<uint16_t*>(reinterpret_cast<const uint16_t*>(&number)));
-                }
-                return tensor;
-                break;
-            }
-            case ovms::Precision::U32: {
-                ov::Tensor tensor = makeTensor(requestInput, tensorInfo);
-                uint32_t* ptr = reinterpret_cast<uint32_t*>(tensor.data());
-                size_t i = 0;
-                for (auto& number : requestInput.contents().uint_contents()) {
-                    ptr[i++] = *(const_cast<uint32_t*>(reinterpret_cast<const uint32_t*>(&number)));
-                }
-                return tensor;
-                break;
-            }
-                // uint64_contents
-            case ovms::Precision::U64: {
-                ov::Tensor tensor = makeTensor(requestInput, tensorInfo);
-                uint64_t* ptr = reinterpret_cast<uint64_t*>(tensor.data());
-                size_t i = 0;
-                for (auto& number : requestInput.contents().uint64_contents()) {
-                    ptr[i++] = *(const_cast<uint64_t*>(reinterpret_cast<const uint64_t*>(&number)));
-                }
-                return tensor;
-                break;
-            }
-                // fp32_contents
-            case ovms::Precision::FP32: {
-                ov::Tensor tensor = makeTensor(requestInput, tensorInfo);
-                float* ptr = reinterpret_cast<float*>(tensor.data());
-                size_t i = 0;
-                for (auto& number : requestInput.contents().fp32_contents()) {
-                    ptr[i++] = *(const_cast<float*>(reinterpret_cast<const float*>(&number)));
-                }
-                return tensor;
-                break;
-            }
-                // fp64_contentes
-            case ovms::Precision::FP64: {
-                ov::Tensor tensor = makeTensor(requestInput, tensorInfo);
-                double* ptr = reinterpret_cast<double*>(tensor.data());
-                size_t i = 0;
-                for (auto& number : requestInput.contents().fp64_contents()) {
-                    ptr[i++] = *(const_cast<double*>(reinterpret_cast<const double*>(&number)));
-                }
-                return tensor;
-                break;
-            }
+#define COPY_EACH_VALUE(OVMS_ENUM_TYPE, C_TYPE, CONTENT_PREFIX)                          \
```

Comment:
Wouldn't it be cleaner to move it outside of this function? It completely breaks indentation

Reply:
identation is broken only for the part of macro definition. It is defined near the place where it is used so I prefer this approach.